code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceExposeMapDaoImpl.java

```diff
@@ -92,24 +55,10 @@ public void run() {
                         .and(SERVICE_EXPOSE_MAP.SERVICE_ID.eq(serviceId))
                         .and(SERVICE_EXPOSE_MAP.STATE.in(CommonStatesConstants.ACTIVATING,
                                 CommonStatesConstants.ACTIVE, CommonStatesConstants.REQUESTED)))
```

Comment:
Darren, when we validate if the service number of instances against the scale (to see if more instances need to be added/removed), we use this method. So as the map gets scheduled for creation as a part of service instance create, not executed, its possible that it might be in Requested state even though the service instance record is created (especially when idempotent checks are on)